### ISSUE TITLE
cy.request Is Not Providing Content-Type Header with Request When followRedirect is true

### ISSUE THREAD

**Issue by todd-m-kemp** [cypress-io/cypress#8906]
### Current behavior

Using `cy.request` with `followRedirect: true` results in a `Content-Type` header not being included with the original request.

### Desired behavior

All specified headers are sent with the request regardless of whether or not the request uses `followRedirect: true` or `followRedirect: false`.

### Test code to reproduce

```
context('Issues with Content-Type Header', function () {

  it('Test 1 - Uses followRedirect: false - Passes Content-Type Header As Expected', function () {
    cy.request({
      method: 'GET',
      url: 'http://secure.vidyard.com/api/v1/features',
      headers: {
        'Content-Type': 'application/json',
      },
      followRedirect: false,
    });
  });

  it('Test 2 - Uses followRedirect: true - Does Not Pass Content-Type Header', function () {
    cy.request({
      method: 'GET',
      url: 'http://secure.vidyard.com/api/v1/features',
      headers: {
        'Content-Type': 'application/json',
      },
      followRedirect: true,
    });
  });
});
```

Both tests pass. However, if you look at the headers sent with the request in each of the two tests, you will see that the `Content-Type` header is listed as a request header for the first test

<img width="718" alt="Untitled" src="https://user-images.githubusercontent.com/62152998/96722390-4f767300-137b-11eb-8be9-408fc2d140c1.png">

but it is not listed as a header in the second test

<img width="718" alt="Untitled2" src="https://user-images.githubusercontent.com/62152998/96722654-9c5a4980-137b-11eb-80a5-1c0a1a140796.png">

This problem is preventing me from using `cy.request` for test setup as I need to provide the `Content-Type` header with a request that redirects but the request will not succeed because the `Content-Type` header is missing.

### Versions

Cypress 5.1.0 on macOS 10.15.6 with Chrome 86 and Firefox 81.

**Comment by jennifer-shehane** [comment]
Yes, I can see that the redirected requests do not have the specified Content-Type in the request headers like you've shown. I'm not sure why it behaves this way.

```js
it('Test 1 - Uses followRedirect: false - Passes Content-Type Header As Expected', () => {
  cy.request({
    method: 'GET',
    url: 'http://secure.vidyard.com/api/v1/features',
    headers: {
      'Content-Type': 'application/json',
    },
    followRedirect: false,
  })
})

it('Test 2 - Uses followRedirect: true - Does Not Pass Content-Type Header', () => {
  cy.request({
    method: 'GET',
    url: 'http://secure.vidyard.com/api/v1/features',
    headers: {
      'Content-Type': 'application/json',
    },
    followRedirect: true,
  })
})
```

**Comment by okeshreddy** [comment]
Hi , Same problem for SOAP call .
Passing Content-Type as **"Content-Type":"application/soap+xml;charset=UTF-8;"** but browser sending  **"Content-Type": "application/x-www-form-urlencoded",** . is there any way to stop the default **"Content-Type":"application/soap+xml;charset=UTF-8;

**Comment by okeshreddy** [comment]
@jennifer-shehane Any update on this

**Comment by svenhorner** [comment]
@jennifer-shehane Is there any update on this?

**Comment by hatemeby** [comment]
@jennifer-shehane do we have any solution for that issue? Or, maybe, some tweaks on how to handle it before it will be fixed?

**Comment by hatemeby** [comment]
I found a solution for my case and it might be useful for someone. 
If you are trying to send a request body with form data as I tried:
<img width="663" alt="Screenshot 2023-03-28 at 11 59 20" src="https://user-images.githubusercontent.com/50153429/228184995-22cf26e8-5e14-44e7-828a-e24d3c19891e.png">
That won't work due to the form = true parameter inside the method:
<img width="941" alt="Screenshot 2023-03-28 at 10 56 32" src="https://user-images.githubusercontent.com/50153429/228185286-b3eb05f7-724f-4c6e-ab4a-0241285f46c2.png">
So Cyress set "application/x-www-form-urlencoded" Content-Type by default. 

In my case, I set form = false, which allows me to change Content-Type to any I need, and create a new FormData() variable. 
![Uploading Screenshot 2023-03-28 at 12.05.23.png…]()
So it will work as expected.

**Comment by adam-thomas-privitar** [comment]
Same issue with `accept` header
